project_report/tex/implementation.tex

```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
+function is used. As the error simulation in the machine is a function of the
+normalised machine speed and the selected product, the normalised machine speed
+must be calculated. This is done by taking 100 over the maximum machine speed.
+This multiplied by chosen machine speed will then give the percentage of the 
+maximum machine speed, which is used when plotting the graph for the optimal
+production speed. The normalised machine speed will be the \(x\)-values and the
+calculated OEE will be the \(y\)-values. An example can be seen in Figure
+\ref{figure:ops_pilsner}, where the graph for the optimal production speed for
+the product type pilsner can be seen.
+
+\begin{figure}[ht]
+	\centering
+	\includegraphics[width=1\linewidth]{images/ops/pilsner.png}
+	\caption{Optimal Production Speed for Pilsner}
+	\label{figure:ops_pilsner}
+\end{figure}
+
+This plot is generated using Microsoft Excel. When the OEE for each tested
+machine speed is plotted, a third degree polynomial trend line is made and an
+expression for the function generated is given. For the pilsner the function is
+given by:
+
+\[y = -0,0003x^3+0,0334x^2-0,1057x+9,0202\]
+
+
+With an \(R^2\) of \(0,9905\) which is used to see how close to the data points
+the polynomial is (the closer to 1 the more accurate).\\
+
+To find the maximum of the function, the derivative is calculated:
+
+\[y' = -\frac{9x^2-668x+1057}{10000}\]
+
+To find the maximum this equation should be solved for \(x\) when \(y=0\). This
+gives \(72,60463\), which is percentage of the maximum machine speed which is
+the most optimal production speed for the pilsner product type. 
+
+To make sure that it is a local maximum at \(72,60463\), the second derivative
+is calculated:
+
+\[y'' = -\frac{18x-668}{10000}\]
+
+Then this equation is solved for \(y\) when \(x=72,60463\), which gives
+\(-0,06388833\). Since the result is less than 0, it can be confirmed that it is
+a maximum.\\
+
+This is done for all product types, and the result can be seen in Table
+\ref{table:ops}.
+
+
+\begin{table}[ht]
+     \begin{tabularx}{\textwidth}{|>{\RaggedRight}p{3cm}|>{\RaggedRight}X|>{\RaggedRight}p{6cm}|}
+     \hline
+     \textbf{Product type} & \textbf{Maximum} & \textbf{Optimal Production Speed}\\
+     \hline
+     Pilsner & 72,61\% of 600 & 435,63 products/minute \\
+     \hline
+     Wheat & 51,58\% of 300 & 154,75 products/minute \\
+     \hline
+     IPA & 62,55\% of 150 & 93,83 products/minute \\
+     \hline
+     Stout & The equation has no solution for \(x\) in ${\rm I\!R}$ & 200 products/minute (max) \\
+     \hline
+     Ale & 91,82\% of 100 & 91,82 products/minute \\
+     \hline
+     Alcohol Free & 76,35\% of 125 & 95,44 products/minute \\
+     \hline
+    \end{tabularx}
+    \caption{Optimal Production Speed}
+    \label{table:ops}
+\end{table}
+
+
+The reason for the result of the stout is, that the function has no local maxima
+in the range (0 to 200) but it keeps increasing, as seen in the graph for the
```

Comment:
Maybe split this sentence into two:

"... in the range (0 to 200). Instead, it keeps increasing, as seen..."